kik_unofficial/client.py

```diff
@@ -160,81 +177,69 @@ def register(self, email, username, password, first_name, last_name, birthday="1
         """
         Sends a text chat message to another person or a group with the given JID/username.
 
         :param peer_jid: The Jabber ID for which to send the message (looks like username_ejs@talk.kik.com)
                          If you don't know the JID of someone, you can also specify a kik username here.
         :param message: The actual message body
-        :param bot_mention_jid: If an official bot is referenced, their jid must be embedded as mention for them
-        to respond.
         """
         peer_jid = self.get_jid(peer_jid)
 
-        if self.is_group_jid(peer_jid):
-            self.log.info(f"Sending chat message '{message}' to group '{peer_jid}'...")
-            return self._send_xmpp_element(chatting.OutgoingGroupChatMessage(peer_jid, message, bot_mention_jid))
-        else:
-            self.log.info(f"Sending chat message '{message}' to user '{peer_jid}'...")
-            return self._send_xmpp_element(chatting.OutgoingChatMessage(peer_jid, message, False, bot_mention_jid))
+        chat_message = chatting.OutgoingChatMessage(peer_jid, message)
+        self.log.info(f"Sending chat message '{message}' to {'group' if chat_message.is_group else 'chat'} '{peer_jid}'...")
+        return self._send_xmpp_element(chat_message)
 
-    def send_chat_image(self, peer_jid: str, file, forward=True):
+    def send_chat_image(self, peer_jid: str, file, forward: bool = True):
         """
         Sends an image chat message to another person or a group with the given JID/username.
         :param peer_jid: The Jabber ID for which to send the message (looks like username_ejs@talk.kik.com)
                          If you don't know the JID of someone, you can also specify a kik username here.
         :param file: The path to the image file OR its bytes OR an IOBase object to send.
+        :param forward: True to allow the client to forward the image to other chats
         """
         peer_jid = self.get_jid(peer_jid)
+        image = chatting.OutgoingChatImage(peer_jid, file, forward)
+        self.log.info(f"Sending chat image to {'group' if image.is_group else 'user'} '{peer_jid}'...")
 
-        if self.is_group_jid(peer_jid):
-            self.log.info(f"Sending chat image to group '{peer_jid}'...")
-            imageRequest = chatting.OutgoingGroupChatImage(peer_jid, file, forward)
-        else:
-            self.log.info(f"Sending chat image to user '{peer_jid}'...")
-            imageRequest = chatting.OutgoingChatImage(peer_jid, file, False, forward)
         content.upload_gallery_image(
-            imageRequest,
+            image,
             f'{self.kik_node}@talk.kik.com',
             self.username,
             self.password,
         )
-        return self._send_xmpp_element(imageRequest)
+        return self._send_xmpp_element(image)
 
-    def send_read_receipt(self, peer_jid: str, receipt_message_id: str, group_jid=None):
+    def send_read_receipt(self, message: XMPPResponse):
```

Comment:
Are you sure this is better than passing the message ID like it was done previously? Is it worth the API break?

Reply:
It is simpler and less error prone. But I see utility in keeping the old method. Is it acceptable to keep two methods, one with the old method signature and the new one? Or is that bad practice?

Reply:
Yes, let's do it, but make sure one is calling the other one, and there is a comments that explains why there are two methods (compatibility and comfort)

Reply:
Resolved in fad996edc257eb8042c08cdc521368174c3fafde